index.js

```diff
@@ -106,6 +107,13 @@ module.exports = function serialize(obj, options) {
             if(origValue instanceof Set) {
                 return '@__S-' + UID + '-' + (sets.push(origValue) - 1) + '__@';
             }
+
+            if(origValue instanceof Array) {
+                var isSparse = origValue.filter(() => true).length != origValue.length;
```

Comment:
I would use non-arrow function for legacy runtime doesn’t know ES6 syntax, and I prefer use `!==` instead of `!=`.
```suggestion
                var isSparse = origValue.filter(function(){return true}).length !== origValue.length;
```